components/prism-robot-framework.js

```diff
@@ -0,0 +1,104 @@
+(function (Prism) {
+
+	var comment = {
+		pattern: /(^[ \t]*| {2}|\t)#.*/m,
+		lookbehind: true,
+		greedy: true
+	};
+
+	var variable = {
+		pattern: /((?:^|[^\\])(?:\\{2})*)[$@&%]\{(?:[^{}\r\n]|\{[^{}\r\n]*\})*\}/,
+		lookbehind: true,
+		inside: {
+			'punctuation': /^[$@&%]\{|\}$/
+		}
+	};
+
+	function createSection(name, inside) {
+		var extendecInside = {};
+
+		extendecInside['section-header'] = {
+			pattern: /^ ?\*{3}.+?\*{3}/,
+			alias: 'keyword'
+		};
+
+		// copy inside tokens
+		for (var token in inside) {
+			extendecInside[token] = inside[token];
+		}
+
+		extendecInside['tag'] = {
+			pattern: /([\r\n](?:  |\t)[ \t]*)\[[-\w]+\]/,
+			lookbehind: true,
+			inside: {
+				'punctuation': /\[|\]/
+			}
+		};
+		extendecInside['variable'] = variable;
+		extendecInside['comment'] = comment;
+
+		return {
+			pattern: RegExp(/^ ?\*{3}[ \t]*<name>[ \t]*\*{3}(?:.|[\r\n](?!\*{3}))*/.source.replace(/<name>/g, name), 'im'),
```

Comment:
Will this work with the gulp task we have in the build that does the `.source` replace?

Reply:
Yes. Here's the minified version of that line (pulled from `components/prism-robot-framework.min.js`):

```js
pattern:RegExp("^ ?\\*{3}[ \t]*<name>[ \t]*\\*{3}(?:.|[\r\\n](?!\\*{3}))*".replace(/<name>/g,t),"im")
```

What made you think it wouldn't work?

Reply:
I didn't remember how it worked and I was wondering if the dynamic rewrite was an issue.

Reply:
> I didn't remember how it worked

Barely. I used a regex to find all RegExp literals which end with `/.source` and replace those with the optimized source value. I say optimized but it's really just a few things like replace `\\t` with `\t` to save a few characters. It's technically incorrect as it's not the real source value but it works well for Prism where we never leave the world of regex patterns.

I'm planning to drop this as soon as our minifier supports this. With regard to #1578, I opened [an issue](https://github.com/terser/terser/issues/400) at [Terser](https://github.com/terser/terser) which can minify ES6+ code.